theano/tensor/slinalg.py

```diff
@@ -97,9 +122,14 @@ def conjugate_solve_triangular(outer, inner):
             chol_x, tril_and_halve_diagonal(chol_x.T.dot(dz)))
 
         if self.lower:
-            return [tensor.tril(s + s.T) - tensor.diag(tensor.diagonal(s))]
+            grad = tensor.tril(s + s.T) - tensor.diag(tensor.diagonal(s))
+        else:
+            grad = tensor.triu(s + s.T) - tensor.diag(tensor.diagonal(s))
+
+        if self.on_error == 'nan':
+            return [tensor.switch(ok, grad, np.nan)]
```

Comment:
if ok is false, this won't return data in the right shape. I think you want to replace np.nan to:

theano.tensor.fill(grad, np.nan)